src/topp/MzTabExporter.cpp

```diff
@@ -194,315 +195,676 @@ namespace OpenMS
-          // select accession of first peptide_evidence as representative ("leading") accession
-          row.accession = peptide_evidences.empty() ? MzTabString("null") : MzTabString(peptide_evidences[0].getProteinAccession());
-          row.best_search_engine_score[1] = MzTabDouble(best_ph.getScore());
-          row.search_engine_score_ms_run[1][1] = MzTabDouble(best_ph.getScore());
+        // fill opt_ column of psm
+        vector<String> ph_keys;
+        best_ph.getKeys(ph_keys);
+        for (Size k = 0; k != ph_keys.size(); ++k)
+        {
+          const String& key = ph_keys[k];
 
-          // find opt_global_modified_sequence in opt_ and set it to the OpenMS amino acid string (easier human readable than unimod accessions)
+          // find matching entry in opt_ (TODO: speed this up)
           for (Size i = 0; i != row.opt_.size(); ++i)
           {
             MzTabOptionalColumnEntry& opt_entry = row.opt_[i];
 
-            if (opt_entry.first == String("opt_global_modified_sequence"))
+            if (opt_entry.first == String("opt_psm_") + key)
             {
-              opt_entry.second = MzTabString(aas.toString());
+              opt_entry.second = MzTabString(best_ph.getMetaValue(key).toString());
             }
           }
+        }
+        rows.push_back(row);
+      }
 
-          // fill opt_ column of psm
-          vector<String> ph_keys;
-          best_ph.getKeys(ph_keys);
-          for (Size k = 0; k != ph_keys.size(); ++k)
-          {
-            const String& key = ph_keys[k];
+      mztab.setPSMSectionRows(rows);
 
-            // find matching entry in opt_ (TODO: speed this up)
-            for (Size i = 0; i != row.opt_.size(); ++i)
-            {
-              MzTabOptionalColumnEntry& opt_entry = row.opt_[i];
+      return mztab;
+    }
+ 
+    static MzTab exportConsensusMapToMzTab(const ConsensusMap& consensus_map, const String& filename)
+    {
+      MzTab mztab;
+      vector<ProteinIdentification> prot_ids = consensus_map.getProteinIdentifications();
+      vector<String> var_mods, fixed_mods;
+      MzTabString db, db_version;
+      if (!prot_ids.empty())
+      {
+        ProteinIdentification::SearchParameters sp = prot_ids[0].getSearchParameters();
+        var_mods = sp.variable_modifications;
+        fixed_mods = sp.fixed_modifications;
+        db = sp.db.empty() ? MzTabString() : MzTabString(sp.db);
+        db_version = sp.db_version.empty() ? MzTabString() : MzTabString(sp.db_version);
+      }
 
-              if (opt_entry.first == String("opt_psm_") + key)
-              {
-                opt_entry.second = MzTabString(best_ph.getMetaValue(key).toString());
-              }
-            }
+      // currently we don't save how many channels we saved in a consensus feature so we scan the file
+      // for the maximum max index in a consensus feature and take this as number of study variables
+      Size n_study_variables = 0;
```

Comment:
Hmm I guess getFileDescriptions is on the ms run level but not on the "assay" or "study variable" level. I will take a look at it - maybe it can be simplified - thanks for pointing this out.